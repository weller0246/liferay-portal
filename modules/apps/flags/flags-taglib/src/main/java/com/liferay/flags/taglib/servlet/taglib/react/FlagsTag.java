/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.flags.taglib.servlet.taglib.react;

import com.liferay.flags.taglib.internal.servlet.ServletContextUtil;
import com.liferay.flags.taglib.internal.servlet.taglib.util.FlagsTagUtil;
import com.liferay.portal.kernel.exception.PortalException;
import com.liferay.portal.kernel.json.JSONObject;
import com.liferay.portal.kernel.json.JSONUtil;
import com.liferay.portal.kernel.language.LanguageUtil;
import com.liferay.portal.kernel.log.Log;
import com.liferay.portal.kernel.log.LogFactoryUtil;
import com.liferay.portal.kernel.model.Company;
import com.liferay.portal.kernel.model.User;
import com.liferay.portal.kernel.theme.ThemeDisplay;
import com.liferay.portal.kernel.util.AggregateResourceBundle;
import com.liferay.portal.kernel.util.HashMapBuilder;
import com.liferay.portal.kernel.util.PortalUtil;
import com.liferay.portal.kernel.util.PortletKeys;
import com.liferay.portal.kernel.util.Validator;
import com.liferay.portal.kernel.util.WebKeys;
import com.liferay.portal.language.LanguageResources;
import com.liferay.taglib.util.IncludeTag;
import com.liferay.taglib.util.TagResourceBundleUtil;

import java.util.Map;
import java.util.ResourceBundle;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.jsp.PageContext;

/**
 * @author Ambrín Chaudhary
 * @author Jorge González
 */
public class FlagsTag extends IncludeTag {

	public String getClassName() {
		return _className;
	}

	public long getClassPK() {
		return _classPK;
	}

	public String getContentTitle() {
		return _contentTitle;
	}

	public String getContentURL() {
		return _contentURL;
	}

	public String getElementClasses() {
		return _elementClasses;
	}

	public String getMessage() {
		return _message;
	}

	public long getReportedUserId() {
		return _reportedUserId;
	}

	public boolean isEnabled() {
		return _enabled;
	}

	public boolean isLabel() {
		return _label;
	}

	public void setClassName(String className) {
		_className = className;
	}

	public void setClassPK(long classPK) {
		_classPK = classPK;
	}

	public void setContentTitle(String contentTitle) {
		_contentTitle = contentTitle;
	}

	public void setContentURL(String contentURL) {
		_contentURL = contentURL;
	}

	public void setElementClasses(String elementClasses) {
		_elementClasses = elementClasses;
	}

	public void setEnabled(boolean enabled) {
		_enabled = enabled;
	}

	public void setLabel(boolean label) {
		_label = label;
	}

	public void setMessage(String message) {
		_message = message;
	}

	@Override
	public void setPageContext(PageContext pageContext) {
		super.setPageContext(pageContext);

		setServletContext(ServletContextUtil.getServletContext());
	}

	public void setReportedUserId(long reportedUserId) {
		_reportedUserId = reportedUserId;
	}

	@Override
	protected void cleanUp() {
		super.cleanUp();

		_className = null;
		_classPK = 0;
		_contentTitle = null;
		_contentURL = null;
		_elementClasses = null;
		_enabled = true;
		_label = true;
		_message = null;
		_reportedUserId = 0;
	}

	@Override
	protected String getPage() {
		return _PAGE;
	}

	@Override
	protected void setAttributes(HttpServletRequest httpServletRequest) {
		try {
			String message = _getMessage();

			httpServletRequest.setAttribute(
				"liferay-flags:flags:data", _getData(message));

			httpServletRequest.setAttribute(
				"liferay-flags:flags:elementClasses", _elementClasses);
			httpServletRequest.setAttribute(
				"liferay-flags:flags:message", message);
			httpServletRequest.setAttribute(
				"liferay-flags:flags:onlyIcon", !_label);
		}
		catch (Exception exception) {
			_log.error(exception, exception);
		}
	}

	private Map<String, Object> _getData(String message)
		throws PortalException {

		return HashMapBuilder.<String, Object>put(
			"context",
			HashMapBuilder.<String, Object>put(
				"namespace", PortalUtil.getPortletNamespace(PortletKeys.FLAGS)
			).build()
		).put(
			"props",
			() -> {
				HttpServletRequest httpServletRequest = getRequest();

				ThemeDisplay themeDisplay =
					(ThemeDisplay)httpServletRequest.getAttribute(
						WebKeys.THEME_DISPLAY);

				return HashMapBuilder.<String, Object>put(
					"baseData", _getDataJSONObject(themeDisplay)
				).put(
					"captchaURI", FlagsTagUtil.getCaptchaURI(httpServletRequest)
				).put(
					"companyName",
					() -> {
						Company company = themeDisplay.getCompany();

						return company.getName();
					}
				).put(
					"disabled", !_enabled
				).put(
					"forceLogin", !FlagsTagUtil.isFlagsEnabled(themeDisplay)
				).put(
					"message",
					() -> {
						if (Validator.isNotNull(message)) {
							return message;
						}

						return null;
					}
				).put(
					"onlyIcon", !_label
				).put(
					"pathTermsOfUse",
					PortalUtil.getPathMain() + "/portal/terms_of_use"
				).put(
					"reasons",
					FlagsTagUtil.getReasons(
						themeDisplay.getCompanyId(), httpServletRequest)
				).put(
					"signedIn", themeDisplay.isSignedIn()
				).put(
					"uri", FlagsTagUtil.getURI(httpServletRequest)
				).build();
			}
		).build();
	}

	private JSONObject _getDataJSONObject(ThemeDisplay themeDisplay) {
		String namespace = PortalUtil.getPortletNamespace(PortletKeys.FLAGS);

		String contentURL = _contentURL;

		if (Validator.isNull(contentURL)) {
			contentURL = FlagsTagUtil.getCurrentURL(getRequest());
		}

		JSONObject dataJSONObject = JSONUtil.put(
			namespace + "className", _className
		).put(
			namespace + "classPK", _classPK
		).put(
			namespace + "contentTitle", _contentTitle
		).put(
			namespace + "contentURL", contentURL
		).put(
			namespace + "reportedUserId", _reportedUserId
		);

		if (themeDisplay.isSignedIn()) {
			User user = themeDisplay.getUser();

			dataJSONObject.put(
				namespace + "reporterEmailAddress", user.getEmailAddress());
		}

		return dataJSONObject;
	}

	private String _getMessage() {
		ResourceBundle resourceBundle = new AggregateResourceBundle(
			TagResourceBundleUtil.getResourceBundle(pageContext),
			LanguageResources.getResourceBundle(
				PortalUtil.getLocale(getRequest())));

		if (Validator.isNotNull(_message)) {
			return LanguageUtil.get(resourceBundle, _message);
		}

		return LanguageUtil.get(resourceBundle, "report");
	}

	private static final String _PAGE = "/flags/page.jsp";

	private static final Log _log = LogFactoryUtil.getLog(FlagsTag.class);

	private String _className;
	private long _classPK;
	private String _contentTitle;
	private String _contentURL;
	private String _elementClasses;
	private boolean _enabled = true;
	private boolean _label = true;
	private String _message;
	private long _reportedUserId;

}