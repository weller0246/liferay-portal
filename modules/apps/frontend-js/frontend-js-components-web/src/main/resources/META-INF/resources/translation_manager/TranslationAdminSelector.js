/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import ClayButton from '@clayui/button';
import ClayDropDown from '@clayui/drop-down';
import ClayIcon from '@clayui/icon';
import ClayLabel from '@clayui/label';
import ClayLayout from '@clayui/layout';
import PropTypes from 'prop-types';
import React, {useEffect, useMemo, useState} from 'react';

import TranslationAdminModal from './TranslationAdminModal';

// These variables are defined here, out of the component, to avoid
// unexpected re-renders

const emptyArray = [];
const noop = () => {};

const TranslationAdminSelector = ({
	activeLanguageIds: initialActiveLanguageIds = emptyArray,
	adminMode,
	ariaLabels = {
		default: Liferay.Language.get('default'),
		manageTranslations: Liferay.Language.get('manage-translations'),
		notTranslated: Liferay.Language.get('not-translated'),
		translated: Liferay.Language.get('translated'),
	},
	availableLocales = emptyArray,
	defaultLanguageId,
	onActiveLanguageIdsChange = noop,
	onSelectedLanguageIdChange = noop,
	selectedLanguageId: initialSelectedLanguageId,
	showOnlyFlags,
	small = false,
	translations = {},
}) => {
	const [activeLanguageIds, setActiveLanguageIds] = useState(
		initialActiveLanguageIds
	);
	const [selectedLanguageId, setSelectedLanguageId] = useState(
		initialSelectedLanguageId
	);
	const [selectorDropdownActive, setSelectorDropdownActive] = useState(false);
	const [translationModalVisible, setTranslationModalVisible] = useState(
		false
	);

	const handleCloseTranslationModal = (activeLanguageIds) => {
		setActiveLanguageIds(activeLanguageIds);

		if (!activeLanguageIds.includes(selectedLanguageId)) {
			setSelectedLanguageId(defaultLanguageId);
		}

		setTranslationModalVisible(false);
	};

	const activeLocales = useMemo(
		() =>
			availableLocales.filter((availableLocale) =>
				activeLanguageIds.includes(availableLocale.id)
			),
		[availableLocales, activeLanguageIds]
	);

	const selectedLocale = useMemo(() => {
		const id = selectedLanguageId ?? defaultLanguageId;

		return availableLocales.find(
			(availableLocale) => availableLocale.id === id
		);
	}, [availableLocales, defaultLanguageId, selectedLanguageId]);

	useEffect(() => {
		onActiveLanguageIdsChange(activeLanguageIds);
	}, [activeLanguageIds, onActiveLanguageIdsChange]);

	useEffect(() => {
		onSelectedLanguageIdChange(selectedLanguageId);
	}, [selectedLanguageId, onSelectedLanguageIdChange]);

	useEffect(() => {
		setActiveLanguageIds(initialActiveLanguageIds);
	}, [initialActiveLanguageIds]);

	useEffect(() => {
		setSelectedLanguageId(initialSelectedLanguageId);
	}, [initialSelectedLanguageId]);

	return (
		<>
			<TranslationAdminModal
				activeLanguageIds={activeLanguageIds}
				ariaLabels={ariaLabels}
				availableLocales={availableLocales}
				defaultLanguageId={defaultLanguageId}
				onClose={handleCloseTranslationModal}
				translations={translations}
				visible={translationModalVisible}
			/>

			<ClayDropDown
				active={selectorDropdownActive}
				onActiveChange={setSelectorDropdownActive}
				trigger={
					<ClayButton
						displayType="secondary"
						monospaced
						small={small}
						title={Liferay.Language.get(
							'select-translation-language'
						)}
					>
						<span className="inline-item">
							<ClayIcon symbol={selectedLocale.symbol} />
						</span>
						<span className="btn-section">
							{selectedLocale.label}
						</span>
					</ClayButton>
				}
			>
				<ClayDropDown.ItemList>
					{activeLocales.map((activeLocale) => {
						const label = activeLocale.label;

						const isDefaultLocale =
							activeLocale.id === defaultLanguageId;

						const localeValue = translations[label];

						return (
							<ClayDropDown.Item
								key={activeLocale.id}
								onClick={() => {
									setSelectedLanguageId(activeLocale.id);
								}}
							>
								<ClayLayout.ContentRow containerElement="span">
									<ClayLayout.ContentCol
										containerElement="span"
										expand
									>
										<ClayLayout.ContentSection>
											<ClayIcon
												className="inline-item inline-item-before"
												symbol={activeLocale.symbol}
											/>

											{label}
										</ClayLayout.ContentSection>
									</ClayLayout.ContentCol>

									{!showOnlyFlags && (
										<ClayLayout.ContentCol containerElement="span">
											<ClayLayout.ContentSection>
												<ClayLabel
													displayType={
														isDefaultLocale
															? 'info'
															: localeValue
															? 'success'
															: 'warning'
													}
												>
													{isDefaultLocale
														? ariaLabels.default
														: localeValue
														? ariaLabels.translated
														: ariaLabels.notTranslated}
												</ClayLabel>
											</ClayLayout.ContentSection>
										</ClayLayout.ContentCol>
									)}
								</ClayLayout.ContentRow>
							</ClayDropDown.Item>
						);
					})}

					{adminMode && (
						<>
							<ClayDropDown.Divider />
							<ClayDropDown.Item
								data-testid="translation-modal-trigger"
								onClick={() => setTranslationModalVisible(true)}
							>
								<ClayLayout.ContentRow containerElement="span">
									<ClayLayout.ContentCol
										containerElement="span"
										expand
									>
										<ClayLayout.ContentSection>
											<ClayIcon
												className="inline-item inline-item-before"
												symbol="automatic-translate"
											/>

											{ariaLabels.manageTranslations}
										</ClayLayout.ContentSection>
									</ClayLayout.ContentCol>
								</ClayLayout.ContentRow>
							</ClayDropDown.Item>
						</>
					)}
				</ClayDropDown.ItemList>
			</ClayDropDown>
		</>
	);
};

TranslationAdminSelector.propTypes = {
	activeLanguageIds: PropTypes.arrayOf(PropTypes.string),
	adminMode: PropTypes.bool,
	ariaLabels: PropTypes.shape({
		default: PropTypes.string,
		manageTranslations: PropTypes.string,
		managementToolbar: PropTypes.string,
		notTranslated: PropTypes.string,
		tranlated: PropTypes.string,
	}),
	availableLocales: PropTypes.arrayOf(PropTypes.object).isRequired,
	defaultLanguageId: PropTypes.string.isRequired,
	showOnlyFlags: PropTypes.bool,
	small: PropTypes.bool,
	translations: PropTypes.object,
};

export default TranslationAdminSelector;
