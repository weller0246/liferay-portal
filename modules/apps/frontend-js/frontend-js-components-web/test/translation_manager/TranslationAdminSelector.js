/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import {act, cleanup, fireEvent, render} from '@testing-library/react';
import React from 'react';

import '@testing-library/jest-dom/extend-expect';

import TranslationAdminSelector from '../../src/main/resources/META-INF/resources/translation_manager/TranslationAdminSelector';

const activeLanguageIds = ['en_US', 'ca_ES'];

const availableLocales = [
	{
		displayName: 'English (United States)',
		id: 'en_US',
		label: 'en-US',
		symbol: 'en-us',
	},
	{
		displayName: 'Arabic (Saudi Arabia)',
		id: 'ar_SA',
		label: 'ar-SA',
		symbol: 'ar-sa',
	},
	{
		displayName: 'Catalan (Spain)',
		id: 'ca_ES',
		label: 'ca-ES',
		symbol: 'ca-es',
	},
	{
		displayName: 'Chinese (China)',
		id: 'zh_CN',
		label: 'zh-CN',
		symbol: 'zh-cn',
	},
	{
		displayName: 'Dutch (Netherlands)',
		id: 'nl_NL',
		label: 'nl-NL',
		symbol: 'nl-nl',
	},
	{
		displayName: 'Finnish (Finland)',
		id: 'fi_FI',
		label: 'fi-FI',
		symbol: 'fi-fi',
	},
	{
		displayName: 'French (France)',
		id: 'fr_FR',
		label: 'fr-FR',
		symbol: 'fr-fr',
	},
	{
		displayName: 'German (Germany)',
		id: 'de_DE',
		label: 'de-DE',
		symbol: 'de-de',
	},
	{
		displayName: 'Hungarian (Hungary)',
		id: 'hu_HU',
		label: 'hu-HU',
		symbol: 'hu-hu',
	},
	{
		displayName: 'Japanese (Japan)',
		id: 'ja_JP',
		label: 'ja-JP',
		symbol: 'ja-jp',
	},
	{
		displayName: 'Portuguese (Brazil)',
		id: 'pt_BR',
		label: 'pt-BR',
		symbol: 'pt-br',
	},
	{
		displayName: 'Spanish (Spain)',
		id: 'es_ES',
		label: 'es-ES',
		symbol: 'es-es',
	},
	{
		displayName: 'Swedish (Sweden)',
		id: 'sv_SE',
		label: 'sv-SE',
		symbol: 'sv-se',
	},
];

const defaultLanguageId = 'en_US';

const props = {
	activeLanguageIds,
	availableLocales,
	defaultLanguageId,
	onActiveLanguageIdsChange: jest.fn(),
	onSelectedLanguageIdChange: jest.fn(),
};

jest.mock(
	'../../src/main/resources/META-INF/resources/translation_manager/TranslationAdminModal',
	() => {
		return ({onClose, visible}) => {
			const handleOnClose = () => {
				onClose(['ar_SA', 'en_US', 'nl_NL']);
			};

			return (
				<>
					{visible && (
						<div className="modal">
							<button className="close" onClick={handleOnClose}>
								Close
							</button>
						</div>
					)}
				</>
			);
		};
	}
);

const TranslationAdminSelectorWithState = () => {
	const [activeLanguageIds, setActiveLanguageIds] = React.useState(
		props.activeLanguageIds
	);

	return (
		<>
			<button
				data-testid="change-state"
				onClick={() =>
					setActiveLanguageIds(['ar_SA', 'en_US', 'nl_NL'])
				}
			>
				Change State
			</button>

			<TranslationAdminSelector
				{...props}
				activeLanguageIds={activeLanguageIds}
			/>
		</>
	);
};

describe('TranslationAdminSelector', () => {
	afterAll(() => {
		jest.useRealTimers();
	});

	afterEach(() => {
		jest.clearAllTimers();
		jest.restoreAllMocks();
		cleanup();
	});

	beforeAll(() => {
		jest.useFakeTimers();
	});

	it('renders a dropdown trigger with the selected locale flag icon as content', () => {
		const {asFragment} = render(<TranslationAdminSelector {...props} />);

		expect(asFragment()).toMatchSnapshot();
	});

	it('renders an open dropdown with the list of active languages', async () => {
		const {getByTitle} = render(<TranslationAdminSelector {...props} />);

		const trigger = getByTitle('select-translation-language');

		fireEvent.click(trigger);

		const dropdownMenu = document.querySelector('.dropdown-menu');

		expect(dropdownMenu).toMatchSnapshot();
	});

	it('renders an open dropdown with the list of active languages and manage translation option', async () => {
		const {getByTitle} = render(
			<TranslationAdminSelector adminMode={true} {...props} />
		);

		const trigger = getByTitle('select-translation-language');

		fireEvent.click(trigger);

		const dropdownMenu = document.querySelector('.dropdown-menu');

		expect(dropdownMenu).toMatchSnapshot();
	});

	it('renders a modal when click on Manage Translations', async () => {
		const {getByTitle} = render(
			<TranslationAdminSelector adminMode={true} {...props} />
		);

		const trigger = getByTitle('select-translation-language');

		fireEvent.click(trigger);

		const dropdownMenu = document.querySelector('.dropdown-menu');

		const manageTranslationsTrigger = dropdownMenu.querySelector(
			'[data-testid="translation-modal-trigger"]'
		);

		fireEvent.click(manageTranslationsTrigger);

		act(() => {
			jest.runAllTimers();
		});

		const modal = document.querySelector('.modal');

		expect(modal).toMatchSnapshot();
	});

	it('renders an open dropdown with updated active locales when added from the Manage Translations Modal', async () => {
		const {getByTitle} = render(
			<TranslationAdminSelector adminMode={true} {...props} />
		);

		const trigger = getByTitle('select-translation-language');

		fireEvent.click(trigger);

		const dropdownMenu = document.querySelector('.dropdown-menu');

		const manageTranslationsTrigger = dropdownMenu.querySelector(
			'[data-testid="translation-modal-trigger"]'
		);

		fireEvent.click(manageTranslationsTrigger);

		act(() => {
			jest.runAllTimers();
		});

		const modalCloseButton = document.querySelector('.modal .close');

		fireEvent.click(modalCloseButton);

		act(() => {
			jest.runAllTimers();
		});

		fireEvent.click(trigger);

		act(() => {
			jest.runAllTimers();
		});

		expect(dropdownMenu).toMatchSnapshot();
		expect(props.onActiveLanguageIdsChange).toHaveBeenLastCalledWith([
			'ar_SA',
			'en_US',
			'nl_NL',
		]);
	});

	it('calls onSelectedLocaleChange callback on dropdown locale selection', () => {
		const {getByTitle} = render(<TranslationAdminSelector {...props} />);

		const trigger = getByTitle('select-translation-language');

		fireEvent.click(trigger);

		act(() => {
			jest.runAllTimers();
		});

		const dropdownMenu = document.querySelector('.dropdown-menu');

		const localeElement = dropdownMenu.querySelectorAll(
			'.dropdown-item'
		)[1];

		fireEvent.click(localeElement);

		act(() => {
			jest.runAllTimers();
		});

		expect(props.onSelectedLanguageIdChange).toHaveBeenLastCalledWith(
			'ca_ES'
		);
	});

	it('is used as a controlled component', () => {
		const {getByTestId, getByTitle} = render(
			<TranslationAdminSelectorWithState />
		);

		const changeStateButton = getByTestId('change-state');

		fireEvent.click(changeStateButton);

		act(() => {
			jest.runAllTimers();
		});

		const trigger = getByTitle('select-translation-language');

		fireEvent.click(trigger);

		act(() => {
			jest.runAllTimers();
		});

		const dropdownMenu = document.querySelector('.dropdown-menu');

		expect(dropdownMenu).toMatchSnapshot();
	});

	it('selects the default locale if the current selected one is removed from the active locales list', () => {
		const {asFragment, getByTitle} = render(
			<TranslationAdminSelector
				adminMode={true}
				selectedLanguageId="ca_ES"
				{...props}
			/>
		);

		const trigger = getByTitle('select-translation-language');

		fireEvent.click(trigger);

		const dropdownMenu = document.querySelector('.dropdown-menu');

		const manageTranslationsTrigger = dropdownMenu.querySelector(
			'[data-testid="translation-modal-trigger"]'
		);

		fireEvent.click(manageTranslationsTrigger);

		act(() => {
			jest.runAllTimers();
		});

		const modalCloseButton = document.querySelector('.modal .close');

		fireEvent.click(modalCloseButton);

		act(() => {
			jest.runAllTimers();
		});

		expect(asFragment()).toMatchSnapshot();
	});
});
