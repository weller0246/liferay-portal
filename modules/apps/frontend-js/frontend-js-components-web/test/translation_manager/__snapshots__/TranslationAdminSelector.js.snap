// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TranslationAdminSelector is used as a controlled component 1`] = `
<div
  class="dropdown-menu show"
  style="left: -999px; top: -995px;"
>
  <ul
    class="list-unstyled"
  >
    <li>
      <button
        class="dropdown-item"
      >
        <span
          class="autofit-row"
        >
          <span
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-section"
            >
              <svg
                class="lexicon-icon lexicon-icon-en-us inline-item inline-item-before"
                role="presentation"
              >
                <use
                  xlink:href="#en-us"
                />
              </svg>
              en-US
            </div>
          </span>
          <span
            class="autofit-col"
          >
            <div
              class="autofit-section"
            >
              <span
                class="label label-info"
              >
                <span
                  class="label-item label-item-expand"
                >
                  default
                </span>
              </span>
            </div>
          </span>
        </span>
      </button>
    </li>
    <li>
      <button
        class="dropdown-item"
      >
        <span
          class="autofit-row"
        >
          <span
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-section"
            >
              <svg
                class="lexicon-icon lexicon-icon-ar-sa inline-item inline-item-before"
                role="presentation"
              >
                <use
                  xlink:href="#ar-sa"
                />
              </svg>
              ar-SA
            </div>
          </span>
          <span
            class="autofit-col"
          >
            <div
              class="autofit-section"
            >
              <span
                class="label label-warning"
              >
                <span
                  class="label-item label-item-expand"
                >
                  not-translated
                </span>
              </span>
            </div>
          </span>
        </span>
      </button>
    </li>
    <li>
      <button
        class="dropdown-item"
      >
        <span
          class="autofit-row"
        >
          <span
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-section"
            >
              <svg
                class="lexicon-icon lexicon-icon-nl-nl inline-item inline-item-before"
                role="presentation"
              >
                <use
                  xlink:href="#nl-nl"
                />
              </svg>
              nl-NL
            </div>
          </span>
          <span
            class="autofit-col"
          >
            <div
              class="autofit-section"
            >
              <span
                class="label label-warning"
              >
                <span
                  class="label-item label-item-expand"
                >
                  not-translated
                </span>
              </span>
            </div>
          </span>
        </span>
      </button>
    </li>
  </ul>
</div>
`;

exports[`TranslationAdminSelector renders a dropdown trigger with the selected locale flag icon as content 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <button
      class="dropdown-toggle btn btn-monospaced btn-secondary"
      style=""
      title="select-translation-language"
      type="button"
    >
      <span
        class="inline-item"
      >
        <svg
          class="lexicon-icon lexicon-icon-en-us"
          role="presentation"
        >
          <use
            xlink:href="#en-us"
          />
        </svg>
      </span>
      <span
        class="btn-section"
      >
        en-US
      </span>
    </button>
  </div>
</DocumentFragment>
`;

exports[`TranslationAdminSelector renders a modal when click on Manage Translations 1`] = `
<div
  class="modal"
>
  <button
    class="close"
  >
    Close
  </button>
</div>
`;

exports[`TranslationAdminSelector renders an open dropdown with the list of active languages 1`] = `
<div
  class="dropdown-menu show"
  style="left: -999px; top: -995px;"
>
  <ul
    class="list-unstyled"
  >
    <li>
      <button
        class="dropdown-item"
      >
        <span
          class="autofit-row"
        >
          <span
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-section"
            >
              <svg
                class="lexicon-icon lexicon-icon-en-us inline-item inline-item-before"
                role="presentation"
              >
                <use
                  xlink:href="#en-us"
                />
              </svg>
              en-US
            </div>
          </span>
          <span
            class="autofit-col"
          >
            <div
              class="autofit-section"
            >
              <span
                class="label label-info"
              >
                <span
                  class="label-item label-item-expand"
                >
                  default
                </span>
              </span>
            </div>
          </span>
        </span>
      </button>
    </li>
    <li>
      <button
        class="dropdown-item"
      >
        <span
          class="autofit-row"
        >
          <span
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-section"
            >
              <svg
                class="lexicon-icon lexicon-icon-ca-es inline-item inline-item-before"
                role="presentation"
              >
                <use
                  xlink:href="#ca-es"
                />
              </svg>
              ca-ES
            </div>
          </span>
          <span
            class="autofit-col"
          >
            <div
              class="autofit-section"
            >
              <span
                class="label label-warning"
              >
                <span
                  class="label-item label-item-expand"
                >
                  not-translated
                </span>
              </span>
            </div>
          </span>
        </span>
      </button>
    </li>
  </ul>
</div>
`;

exports[`TranslationAdminSelector renders an open dropdown with the list of active languages and manage translation option 1`] = `
<div
  class="dropdown-menu show"
  style="left: -999px; top: -995px;"
>
  <ul
    class="list-unstyled"
  >
    <li>
      <button
        class="dropdown-item"
      >
        <span
          class="autofit-row"
        >
          <span
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-section"
            >
              <svg
                class="lexicon-icon lexicon-icon-en-us inline-item inline-item-before"
                role="presentation"
              >
                <use
                  xlink:href="#en-us"
                />
              </svg>
              en-US
            </div>
          </span>
          <span
            class="autofit-col"
          >
            <div
              class="autofit-section"
            >
              <span
                class="label label-info"
              >
                <span
                  class="label-item label-item-expand"
                >
                  default
                </span>
              </span>
            </div>
          </span>
        </span>
      </button>
    </li>
    <li>
      <button
        class="dropdown-item"
      >
        <span
          class="autofit-row"
        >
          <span
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-section"
            >
              <svg
                class="lexicon-icon lexicon-icon-ca-es inline-item inline-item-before"
                role="presentation"
              >
                <use
                  xlink:href="#ca-es"
                />
              </svg>
              ca-ES
            </div>
          </span>
          <span
            class="autofit-col"
          >
            <div
              class="autofit-section"
            >
              <span
                class="label label-warning"
              >
                <span
                  class="label-item label-item-expand"
                >
                  not-translated
                </span>
              </span>
            </div>
          </span>
        </span>
      </button>
    </li>
    <li
      aria-hidden="true"
      class="dropdown-divider"
      role="presentation"
    />
    <li>
      <button
        class="dropdown-item"
        data-testid="translation-modal-trigger"
      >
        <span
          class="autofit-row"
        >
          <span
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-section"
            >
              <svg
                class="lexicon-icon lexicon-icon-automatic-translate inline-item inline-item-before"
                role="presentation"
              >
                <use
                  xlink:href="#automatic-translate"
                />
              </svg>
              manage-translations
            </div>
          </span>
        </span>
      </button>
    </li>
  </ul>
</div>
`;

exports[`TranslationAdminSelector renders an open dropdown with updated active locales when added from the Manage Translations Modal 1`] = `
<div
  class="dropdown-menu"
  style="left: -999px; top: -995px;"
>
  <ul
    class="list-unstyled"
  >
    <li>
      <button
        class="dropdown-item"
      >
        <span
          class="autofit-row"
        >
          <span
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-section"
            >
              <svg
                class="lexicon-icon lexicon-icon-en-us inline-item inline-item-before"
                role="presentation"
              >
                <use
                  xlink:href="#en-us"
                />
              </svg>
              en-US
            </div>
          </span>
          <span
            class="autofit-col"
          >
            <div
              class="autofit-section"
            >
              <span
                class="label label-info"
              >
                <span
                  class="label-item label-item-expand"
                >
                  default
                </span>
              </span>
            </div>
          </span>
        </span>
      </button>
    </li>
    <li>
      <button
        class="dropdown-item"
      >
        <span
          class="autofit-row"
        >
          <span
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-section"
            >
              <svg
                class="lexicon-icon lexicon-icon-ar-sa inline-item inline-item-before"
                role="presentation"
              >
                <use
                  xlink:href="#ar-sa"
                />
              </svg>
              ar-SA
            </div>
          </span>
          <span
            class="autofit-col"
          >
            <div
              class="autofit-section"
            >
              <span
                class="label label-warning"
              >
                <span
                  class="label-item label-item-expand"
                >
                  not-translated
                </span>
              </span>
            </div>
          </span>
        </span>
      </button>
    </li>
    <li>
      <button
        class="dropdown-item"
      >
        <span
          class="autofit-row"
        >
          <span
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-section"
            >
              <svg
                class="lexicon-icon lexicon-icon-nl-nl inline-item inline-item-before"
                role="presentation"
              >
                <use
                  xlink:href="#nl-nl"
                />
              </svg>
              nl-NL
            </div>
          </span>
          <span
            class="autofit-col"
          >
            <div
              class="autofit-section"
            >
              <span
                class="label label-warning"
              >
                <span
                  class="label-item label-item-expand"
                >
                  not-translated
                </span>
              </span>
            </div>
          </span>
        </span>
      </button>
    </li>
    <li
      aria-hidden="true"
      class="dropdown-divider"
      role="presentation"
    />
    <li>
      <button
        class="dropdown-item"
        data-testid="translation-modal-trigger"
      >
        <span
          class="autofit-row"
        >
          <span
            class="autofit-col autofit-col-expand"
          >
            <div
              class="autofit-section"
            >
              <svg
                class="lexicon-icon lexicon-icon-automatic-translate inline-item inline-item-before"
                role="presentation"
              >
                <use
                  xlink:href="#automatic-translate"
                />
              </svg>
              manage-translations
            </div>
          </span>
        </span>
      </button>
    </li>
  </ul>
</div>
`;

exports[`TranslationAdminSelector selects the default locale if the current selected one is removed from the active locales list 1`] = `
<DocumentFragment>
  <div
    class="dropdown"
  >
    <button
      class="dropdown-toggle btn btn-monospaced btn-secondary"
      style=""
      title="select-translation-language"
      type="button"
    >
      <span
        class="inline-item"
      >
        <svg
          class="lexicon-icon lexicon-icon-en-us"
          role="presentation"
        >
          <use
            xlink:href="#en-us"
          />
        </svg>
      </span>
      <span
        class="btn-section"
      >
        en-US
      </span>
    </button>
  </div>
</DocumentFragment>
`;
